tests/unit/via/views/proxy_test.py

```diff
@@ -6,20 +6,16 @@
 
 
 class TestProxy:
-    def test_it(
-        self, pyramid_request, get_url_details, via_client_service, url_from_user_input
-    ):
-        pyramid_request.path = "/https://example.org"
+    def test_it(self, pyramid_request, get_url_details, via_client_service):
+        pyramid_request.path_qs = "/https://example.org?a=1&via.sec=HEX"
```

Comment:
I've changed this test to actually just show what we do with things instead of mocking out the URL manipulation routines. 

The is a more integrated test, but it actually proves it does something rather than just an assertion about what we call. All of these things are stateless.

Reply:
I think this is probably fine given that `url_from_user_input()` is just a nearby helper